### PR TITLE
Add LwMutex functions

### PR DESCRIPTION
**Pull request by sajattack**

### CODE REVIEW

File: psp/src/sys/kernel/thread.rs [hunk 1]
```diff
@@ -950,6 +967,89 @@ psp_extern! {
...
+    ///
+    /// # Return Value
+    ///
+    /// 0 on success, otherwise an error code
+    pub fn sceKernelTryLockLwMutex(
+        work_area: *mut SceLwMutexWorkarea,
+        lock_count: i32,
+    ) -> i32;
+
+    #[psp(0x60107536)]
```
Comment (overdrivenpotato) [inline psp/src/sys/kernel/thread.rs:1020]: Seems that this NID is mistaken, this is for `sceKernelDeleteLwMutex`

File: psp/src/sys/kernel/thread.rs [hunk 2]
```diff
@@ -950,6 +967,89 @@ psp_extern! {
...
+    #[psp(0x19CFF145)]
+    /// Create a lightweight mutex
+    ///
+    /// # Parameters
+    ///
+    /// - `work_area`: The pointer to the workarea
+    /// - `name`: The name of the lightweight mutex
+    /// - `initial_count`: The inital value of the mutex
+    /// - `options_ptr`: Other options for the mutex
+        ///
```
Comment (overdrivenpotato) [inline psp/src/sys/kernel/thread.rs]: Four spaces, not 8

File: psp/src/sys/kernel/thread.rs [hunk 3]
```diff
@@ -950,6 +967,89 @@ psp_extern! {
...
+    /// # Return Value
+    ///
+    /// 0 on success, otherwise an error code
+    pub fn sceKernelCreateLwMutex(
+        work_area: *mut SceLwMutexWorkarea,
+        name: *const u8,
+        attr: u32,
```
Comment (overdrivenpotato) [inline psp/src/sys/kernel/thread.rs:987]: Missing doc comment, what does this represent?

**Review by overdrivenpotato — CHANGES_REQUESTED** [review]

File: psp/src/lib.rs [hunk 4]
```diff
@@ -19,6 +19,8 @@
 // For the `const_generics` feature.
 #![allow(incomplete_features)]
 
+#![recursion_limit = "256"]
```
Comment (sajattack) [inline psp/src/lib.rs:22]: rustc told me to add this because the macros were recursing too hard?

Comment (overdrivenpotato) [reply psp/src/lib.rs:22]: Can this be removed now or is it still necessary?

Comment (sajattack) [reply psp/src/lib.rs:22]: Looks good now.

**Review by overdrivenpotato — APPROVED** [review]

**Pull request merged**